src/engine/ScreenEffect.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Definition for a screen effect
+/// </summary>
+/// <remarks>
+///   <para>
+///     Values for each screen effect
+///   </para>
+/// </remarks>
+public class ScreenEffect : IRegistryType
+{
+    /// <summary>
+    ///   User readable name
+    /// </summary>
+    [TranslateFrom(nameof(untranslatedName))]
+    public string Name = null!;
+
+#pragma warning disable 169,649 // Used through reflection
+    private string? untranslatedName;
+#pragma warning restore 169,649
+
+    /// <summary>
+    ///   Index for this screen effect in the effect menu
+    /// </summary>
+    [JsonProperty]
+    public int Index { get; private set; }
+
+    [JsonProperty]
+    public string? ShaderPath { get; set; }
```

Comment:
```suggestion
    public string? ShaderPath { get; private set; }
```
Is there a reason why the set is public?